src/cmd/logic-function.test.js

```diff
@@ -245,8 +242,417 @@ describe('LogicFunctionCommands', () => {
+			logicFunctions.push(logicFunc2.logic_functions[0]);
+
+			const res = await logicFunctionCommands._getNameFromId('0021e8f4-64ee-416d-83f3-898aa909fb1c', logicFunctions);
+
+			expect(res).to.equal(null);
+		});
+	});
+
+	describe('get', async () => {
+		beforeEach(() => {
+			fs.remove(path.join(process.cwd(), 'LF1'));
+			fs.remove(path.join(process.cwd(), 'LF2'));
+		});
+
+		it('downloads a logic function with name', async () => {
+			let logicFunctions = [];
+			logicFunctions.push(logicFunc1.logic_functions[0]);
+			logicFunctions.push(logicFunc2.logic_functions[0]);
+
+			const stubList = nock('https://api.particle.io/v1',)
+				.intercept('/logic/functions', 'GET')
+				.reply(200, { logic_functions: logicFunctions });
+
+			const stubGet = nock('https://api.particle.io/v1',)
+				.intercept('/logic/functions/0021e8f4-64ee-416d-83f3-898aa909fb1b', 'GET')
+				.reply(200, { logic_function: logicFunc1.logic_functions[0] });
+
+			await logicFunctionCommands.get({ name: 'LF1' });
+
+			expect(fs.existsSync(path.join(process.cwd(), 'LF1', 'LF1.js'))).to.be.true;
+			expect(fs.existsSync(path.join(process.cwd(), 'LF1', 'LF1.logic.json'))).to.be.true;
+			expect(stubList.isDone()).to.be.true;
+			expect(stubGet.isDone()).to.be.true;
+		});
+
+		it('downloads a logic function with id', async () => {
+			let logicFunctions = [];
+			logicFunctions.push(logicFunc1.logic_functions[0]);
+			logicFunctions.push(logicFunc2.logic_functions[0]);
+
+			const stubList = nock('https://api.particle.io/v1',)
+				.intercept('/logic/functions', 'GET')
+				.reply(200, { logic_functions: logicFunctions });
+
+			const stubGet = nock('https://api.particle.io/v1',)
+				.intercept('/logic/functions/0021e8f4-64ee-416d-83f3-898aa909fb1b', 'GET')
+				.reply(200, { logic_function: logicFunc1.logic_functions[0] });
+
+			await logicFunctionCommands.get({ id: '0021e8f4-64ee-416d-83f3-898aa909fb1b' });
+
+			expect(fs.existsSync(path.join(process.cwd(), 'LF1', 'LF1.js'))).to.be.true;
+			expect(fs.existsSync(path.join(process.cwd(), 'LF1', 'LF1.logic.json'))).to.be.true;
+
+			// //clean up
+			fs.remove(path.join(process.cwd(), 'LF1'));
+			expect(stubList.isDone()).to.be.true;
+			expect(stubGet.isDone()).to.be.true;
+		});
+
+		it('downloads a logic function from user input', async () => {
+			let logicFunctions = [];
+			logicFunctions.push(logicFunc1.logic_functions[0]);
+			logicFunctions.push(logicFunc2.logic_functions[0]);
+
+			const stubList = nock('https://api.particle.io/v1',)
+				.intercept('/logic/functions', 'GET')
+				.reply(200, { logic_functions: logicFunctions });
+
+			const stubGet = nock('https://api.particle.io/v1',)
+				.intercept('/logic/functions/0021e8f4-64ee-416d-83f3-898aa909fb1b', 'GET')
+				.reply(200, { logic_function: logicFunc1.logic_functions[0] });
+
+			await logicFunctionCommands.get({ name: 'LF1' });
+
+			expect(fs.existsSync(path.join(process.cwd(), 'LF1', 'LF1.js'))).to.be.true;
```

Comment:
Pretty sure that's making linux tests fail.
I'd rather use `fixtures/__root__/...` folder for creation. Let's try to repro the test error in a linux machine.
the only thing that came to my mind now is `lf1` dir and files should be in lower case since we're converting the name with `slugify` function.